packages/payment-widget/src/lib/components/payment-confirmation.svelte

```diff
@@ -0,0 +1,418 @@
+<script lang="ts">
+  import CloseIcon from "@requestnetwork/shared-icons/close.svelte";
+  import CopyIcon from "@requestnetwork/shared-icons/copy-icon.svelte";
+  import ExchangeIcon from "@requestnetwork/shared-icons/exchange.svelte";
+  import InfoCircleIcon from "@requestnetwork/shared-icons/info-circle.svelte";
+  import type { Web3Modal } from "@web3modal/ethers5";
+  import { onDestroy, onMount } from "svelte";
+  import type { Currency } from "../types";
+  import { chains } from "../utils/chains";
+  import { NETWORK_LABEL } from "../utils/currencies";
+  import {
+    handleRequestPayment,
+    prepareRequestParameters,
+  } from "../utils/request";
+  import Spinner from "./spinner.svelte";
+  import WalletInfo from "./wallet-info.svelte";
+
+  export let selectedCurrency: Currency;
+  export let amountInUSD: number;
+  export let sellerAddress: string;
+  export let currentPaymentStep: string;
+  export let web3Modal: Web3Modal | null;
+  export let isConnected: boolean;
+  export let persistRequest: boolean;
+  export let onPaymentSuccess: (request: any) => void;
+  export let onPaymentError: (error: string) => void;
+  const COUNTDOWN_INTERVAL = 30;
+
+  let amountInCrypto: number = 0;
+  let countdown: number = COUNTDOWN_INTERVAL;
+  let intervalId: NodeJS.Timeout;
+  let exchangeRate: number = 0;
+  let error: string = "";
+
+  $: isLoadingPrice = true;
+  $: isPaying = false;
+
+  const currencySymbol = selectedCurrency.symbol.includes(
+    selectedCurrency.network
+  )
+    ? selectedCurrency.symbol.split("-")[0]
+    : selectedCurrency.symbol;
+
+  async function fetchExchangeRate() {
+    if (isPaying) return;
+    try {
+      isLoadingPrice = true;
+      const response = await fetch(
+        `https://api.coinbase.com/v2/exchange-rates?currency=USD`
+      );
+      const data = await response.json();
+
+      let symbolCurrency = selectedCurrency.symbol.split("-")[0];
+      const symbolMap: { [key: string]: string } = {
+        fUSDC: "USDC",
+        fUSDT: "USDT",
+      };
+      const lookupSymbol = symbolMap[symbolCurrency] || symbolCurrency;
+      const rate = data.data.rates[lookupSymbol];
+      exchangeRate = parseFloat(rate);
+      amountInCrypto = amountInUSD * parseFloat(rate);
```

Comment:
Instead of calculating parseFloat a second time, use the `exchangeRate` variable
```suggestion
      exchangeRate = parseFloat(rate);
      amountInCrypto = amountInUSD * exchangeRate;
```